### PR TITLE
Update rmw_context_impl_t definition

### PR DESCRIPTION
**Pull request by hidmic**
Connected to https://github.com/ros2/rmw/pull/313.

### CODE REVIEW

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 1]
```diff
@@ -267,7 +267,7 @@ struct CddsDomain
   {}
 };
 
-struct rmw_context_impl_t
+struct rmw_context_impl_s
```
Comment (clalancette) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:270]: Same comment as https://github.com/ros2/rmw_fastrtps/pull/558#discussion_r707317216

**Review by eboasson — COMMENTED** [review]
I have not given enough thought to it to be sure it is worth the bother to change all the definitions in the RMW headers, but given that change, this is fine with me.

File: rmw_cyclonedds_cpp/src/rmw_node.cpp (hunk 1 above)
Comment (hidmic) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:271]: See https://github.com/ros2/rmw_cyclonedds/pull/337/commits/67d036b3de6f786ae965671310524a6e7a8bbbd7.

**Review by clalancette — APPROVED** [review]

**Comment by hidmic** [comment]
Full CI (from https://github.com/ros2/rmw/pull/313):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15218/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9938/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12898/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15400/)

**Pull request merged**